File: src/main/java/com/eadms/service/TicketService.java
package com.eadms.service;

import com.eadms.dto.request.TicketCreateRequest;
import com.eadms.dto.request.TicketUpdateRequest;
import com.eadms.dto.response.TicketResponse;
import com.eadms.entity.Ticket;

import java.util.List;

public interface TicketService {
    TicketResponse createTicket(TicketCreateRequest request, Long userId);
    TicketResponse getTicketById(Long id, Long userId, String role);
    List<TicketResponse> getMyTickets(Long userId);
    List<TicketResponse> getAllTickets();
    List<TicketResponse> getTicketsByStatus(Ticket.Status status);
    TicketResponse updateTicketStatus(Long id, TicketUpdateRequest request);
}

File: frontend/src/services/ticketService.ts
import api from './api'
import { Ticket, TicketCreateRequest, TicketUpdateRequest, TicketStatus } from '@/types/ticket'

export const ticketService = {
  async createTicket(request: TicketCreateRequest): Promise<Ticket> {
    const response = await api.post<Ticket>('/tickets', request)
    return response.data
  },

  async getMyTickets(): Promise<Ticket[]> {
    const response = await api.get<Ticket[]>('/tickets/my-tickets')
    return response.data
  },

  async getTicketById(id: number): Promise<Ticket> {
    const response = await api.get<Ticket>(`/tickets/${id}`)
    return response.data
  },

  async getAllTickets(): Promise<Ticket[]> {
    const response = await api.get<Ticket[]>('/tickets/admin/all')
    return response.data
  },

  async getTicketsByStatus(status: TicketStatus): Promise<Ticket[]> {
    const response = await api.get<Ticket[]>(`/tickets/admin/status/${status}`)
    return response.data
  },

  async updateTicketStatus(id: number, request: TicketUpdateRequest): Promise<Ticket> {
    const response = await api.put<Ticket>(`/tickets/admin/${id}`, request)
    return response.data
  },
}
